library/java/net/openid/appauth/AuthorizationManagementActivity.java

```diff
@@ -103,11 +104,18 @@
  *       destroyed, if necessary, such that AuthorizationManagementActivity is once again at the
  *       top of the back stack.
  *
- *     - Step S3: The pending intent provided to
- *       {@link AuthorizationService#performAuthorizationRequest} for completion of the
- *       authorization flow is invoked, providing the decoded {@link AuthorizationResponse} or
- *       {@link AuthorizationException} as appropriate. The AuthorizationManagementActivity
- *       finishes, removing itself from the back stack.
+ *     - Step S3a: If this activity was invoked via
+ *       {@link AuthorizationService#performAuthorizationRequest}, then the pending intent provided
+ *       for completion of the authorization flow is invoked, providing the decoded
+ *       {@link AuthorizationResponse} or {@link AuthorizationException} as appropriate.
+ *       The AuthorizationManagementActivity finishes, removing itself from the back stack.
+ *
+ *     - Step S3b: If this activity was invoked via an intent returned by
+ *       {@link AuthorizationService#getAuthorizationRequestIntent}, then this activity
+ *       calls {@link Activity#setResult(int, Intent)} with {@link Activity#RESULT_OK}
```

Comment:
Only happens if redirect activity is never reached (so not the S flow). It's mentioned in step C2b.